src/main/java/com/sourcery/gymapp/backend/userProfile/service/UserProfileService.java

```diff
@@ -0,0 +1,64 @@
+package com.sourcery.gymapp.backend.userProfile.service;
+
+import com.sourcery.gymapp.backend.userProfile.dto.UserProfileDto;
+import com.sourcery.gymapp.backend.userProfile.exception.UserNotFoundException;
+import com.sourcery.gymapp.backend.userProfile.exception.UserProfileNotFoundException;
+import com.sourcery.gymapp.backend.userProfile.model.UserProfile;
+import com.sourcery.gymapp.backend.userProfile.repository.UserProfileRepository;
+import com.sourcery.gymapp.backend.userProfile.mapper.UserProfileMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class UserProfileService {
+
+    private final ProfileCurrentUserService currentUserService;
+    private final UserProfileRepository userProfileRepository;
+    private final UserProfileMapper userProfileMapper;
+
+    public UserProfileDto getUserProfile() throws UserNotFoundException, UserProfileNotFoundException {
+
+        UUID currentUserId = currentUserService.getCurrentUserId();
+
+        UserProfile userProfile = userProfileRepository
+                .findUserProfileByUserId(currentUserId);
+
+        if (userProfile==null) {
+            throw new UserProfileNotFoundException();
+        }
```

Comment:
UserProfile userProfile = userProfileRepository
                .findUserProfileByUserId(currentUserId);

        if (userProfile==null) {
            throw new UserProfileNotFoundException();
        }
        
        Those lines are repeated in several places -> should be a separate method